ComponentKit/LayoutComponents/CKFlexboxComponent.mm

```diff
@@ -56,13 +58,22 @@ + (instancetype)newWithView:(const CKComponentViewConfiguration &)view
                    children:(CKContainerWrapper<std::vector<CKFlexboxComponentChild>> &&)children
 {
   CKFlexboxComponent * const component = [super newWithView:view size:size];
+
   if (component) {
     component->_style = style;
     component->_children = children.take();
   }
+  
   return component;
 }
 
+- (std::vector<CKComponent *>)render:(id)state
+{
+  return  CK::map(_children, [](CKFlexboxComponentChild child) {
+    return child.component;
```

Comment:
this loses all the flexbox props. who applies them?

Reply:
Or does this just exist to ensure that the children are built by the infrastructure?

Reply:
If so, what happens if I forget to return a child? That's a failure case we should consider.

Reply:
Same as the comment above. The render only creates the component tree, it's not responsible to compute the layout. Means that the `CKFlexboxComponent` works exactly as before. We compute the layout with `computeLayoutThatFits:` and this method hasn't been changed.

Reply:
> what happens if I forget to return a child?

I think it's similar to what we have today - if you create a component in the constructor but doesn't reflect it in your layout it won't be in the component layout, right?
People usually subclass from `CKCompositeComponent` which takes care of the layout for you. However, if you have more than one child you have to implement the `computeLayoutThatFits:` in order to reflect it in the layout.